core/src/bin/generate.ts

```diff
@@ -5,115 +5,124 @@ import {
+      : ""
+  }`;
+  inputs = {
+    list: {
+      required: false,
+      default: false,
+      letter: "l",
+      flag: true,
+      description:
+        'Display the available config templates to use with config-generate.  You can filter the list of templates by providing a string to match against, ie: `grouparoo generate app --list` to see templates which match "app"',
+    },
+    describe: {
+      required: false,
+      default: false,
+      letter: "d",
+      flag: true,
+      description: "Display the options for the template in detail",
+    },
+    parent: {
+      required: false,
+      letter: "p",
+      description:
+        "The id of the object that is the direct parent of this new object, e.g. the appId if you are creating a new Source.",
+    },
+    from: {
+      required: false,
+      letter: "f",
+      description:
+        "For batch generators, where should we read the objects from?",
+    },
+    with: {
+      required: false,
+      letter: "w",
+      description:
+        'For batch generators, what additional objects should we create? Use commas to separate names (--with "id,first_name,last_name") or "*" for all (`--with "*"`). Default is ``',
+      default: "",
+    },
+    mapping: {
+      required: false,
+      letter: "m",
+      description:
+        'For batch generators, how should we map this object? The remote key precedes the Grouparoo Property name. Use = to set the pair (--mapping "id=user_id").',
+    },
+    "high-water-mark": {
+      required: false,
+      letter: "H",
+      description:
+        "For batch generators, what should we use for the high-water-mark?",
+    },
+    "sync-mode": {
+      required: false,
+      letter: "s",
+      description:
+        "For destination generators, how will we sync to the destination?",
+    },
+    overwrite: {
+      required: true,
+      default: false,
+      letter: "o",
+      flag: true,
+      description: "Overwrite existing files?",
+    },
+  };
+  example = `App generation:
+  grouparoo generate postgres:app data_warehouse
+
+Simple Source Generation (needs parent app):
+  grouparoo generate postgres:table:source users_table \\
+    --parent data_warehouse
+
+Batch Source Generation (needs parent app to be applied first):
+  grouparoo generate postgres:table:source users_table \\
+    --parent data_warehouse \\
+    --from users \\
+    --with id,first_name,email,last_name \\
+    --mapping 'id=user_id' \\
+    --high-water-mark updated_at
+
+Learn more about the options for a specific template:
+  grouparoo generate postgres:app --describe`;
 
   preInitialize = () => {
     GrouparooCLI.setGrouparooRunMode(this);
     GrouparooCLI.setNextDevelopmentMode();
   };
 
-  async run({ params }) {
+  async run({
+    params,
+  }: {
+    params: Partial<
+      ParamsFrom<Generate> & {
+        template: string;
+        id: string;
+        path?: string;
+        highWaterMark?: string;
```

Comment:
Commander messes with `-` arg names and camelCases them... that makes typing this odd.

Reply:
Facinating! There are some utilities like this in `type-fest` already - I'll try them out! https://github.com/sindresorhus/type-fest#change-case

Reply:
eh, it doesn't seem to be worth it all the time - there are also other utilities we get from commander, like taking the `no-export` flag and providing the negation `export:boolean` as a param.  I think we'll always need something custom